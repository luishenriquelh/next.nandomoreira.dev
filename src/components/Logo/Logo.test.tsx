import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Logo } from '.';

describe('<Logo />', () => {
  afterEach(cleanup);

  test('Render component', () => {
    const componentTree = render(<Logo />);
    expect(componentTree).toMatchSnapshot();
  });

  test('Render component with href prop', () => {
    const componentTree = render(<Logo href="/" />);
    expect(componentTree).toMatchSnapshot();
  });

  test('Render component with title prop', () => {
    const componentTree = render(<Logo title="Fernando" />);
    expect(componentTree).toMatchSnapshot();
  });
});
