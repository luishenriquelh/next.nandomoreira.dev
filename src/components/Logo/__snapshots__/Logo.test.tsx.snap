// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> Render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  cursor: pointer;
  display: inline-block;
}

<body>
    <div>
      <a
        class="c0"
      >
        <svg
          class=""
          fill="none"
          height="45"
          width="50"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M43.247 0v45H32.662L12.597 21.021V45H0V0h10.584L30.65 23.979V0h12.598z"
          />
          <rect
            class="dot"
            height="12.857"
            rx="6.429"
            width="12.987"
            x="37.013"
            y="32.143"
          />
        </svg>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="Logostyles__Logo-ztqmk5-0 jbrqWl"
    >
      <svg
        class="Logostyles__Svg-ztqmk5-1 FzJFq"
        fill="none"
        height="45"
        width="50"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M43.247 0v45H32.662L12.597 21.021V45H0V0h10.584L30.65 23.979V0h12.598z"
        />
        <rect
          class="dot"
          height="12.857"
          rx="6.429"
          width="12.987"
          x="37.013"
          y="32.143"
        />
      </svg>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
